### PR TITLE
[feature] setup runners for each gem

### PR DESCRIPTION
**Pull request by za4emnik**
https://garage.atlassian.net/browse/PEO-67

### CODE REVIEW

File: inquisition.gemspec [hunk 1]
```diff
@@ -28,11 +28,14 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'simplecov-lcov'
   spec.add_development_dependency 'undercover'
   spec.add_development_dependency 'rubycritic'
+  spec.add_development_dependency 'devise', '~> 3.1.0'
```
Comment (dzemlianoi-double) [inline inquisition.gemspec]: remove

File: inquisition.gemspec [hunk 2]
```diff
@@ -28,11 +28,14 @@ Gem::Specification.new do |spec|
...
+  spec.add_development_dependency 'sidekiq', '~> 0.5.0'
 
   # common
   spec.add_dependency 'awesome_print'        # https://github.com/awesome-print/awesome_print
   spec.add_dependency 'benchmark-ips'        # https://github.com/evanphx/benchmark-ips
   spec.add_dependency 'colorize'             # https://github.com/fazibear/colorize
+  spec.add_dependency 'dry-transaction'
```
Comment (dzemlianoi-double) [inline inquisition.gemspec]: remove

File: inquisition.gemspec [hunk 3]
```diff
@@ -28,11 +28,14 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'simplecov-lcov'
   spec.add_development_dependency 'undercover'
   spec.add_development_dependency 'rubycritic'
+  spec.add_development_dependency 'devise', '~> 3.1.0'
+  spec.add_development_dependency 'sidekiq', '~> 0.5.0'
```
Comment (dzemlianoi-double) [inline inquisition.gemspec]: remove

File: lib/inquisition/auditors/backend/bundler_audit/formatter.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                linterable_name: vulnerability['Name'],
+                linterable_type: linterable_type_default,
+                message: vulnerability['Title'],
+                type: vulnerability['Criticality'],
+                line: line_default
+              }
+            end
+          end
+
+          def data
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/formatter.rb]: rename

File: lib/inquisition/auditors/backend/bundler_audit/formatter.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  module Auditors
+    module Backend
+      module BundlerAudit
+        class Formatter < Core::BaseFormatter
+          private
+
+          def build_errors
+            @build_errors ||= data.map do |vulnerability|
+              {
+                linterable_name: vulnerability['Name'],
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/formatter.rb]: keys to constant

File: lib/inquisition/auditors/backend/bundler_audit/runner.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    module Backend
+      module BundlerAudit
+        class Runner < Auditors::Runner
+          private
+
+          def command
+            'bundle audit'
+          end
+
+          def formatter
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/runner.rb]: try to rethink the logic

File: lib/inquisition/auditors/backend/dawnscanner/formatter.rb [hunk 7]
```diff
@@ -0,0 +1,35 @@
+module Inquisition
+  module Auditors
+    module Backend
+      module Dawnscanner
+        class Formatter < Core::BaseFormatter
+          private
+
+          def build_total_files_count
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/dawnscanner/formatter.rb]: please order your formatters methods similar in each formatter

File: lib/inquisition/auditors/backend/dawnscanner/formatter.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+          def build_error_count
+            data['vulnerabilities_count']
+          end
+
+          def build_errors
+            data['vulnerabilities'].map do |vulnerability|
+              {
+                linterable_name: linterable_name_default,
+                linterable_type: linterable_type_gem,
+                message: vulnerability['message'],
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/dawnscanner/formatter.rb]: constants

File: lib/inquisition/auditors/backend/dawnscanner/runner.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
+module Inquisition
+  module Auditors
+    module Backend
+      module Dawnscanner
+        class Runner < Auditors::Runner
+          private
+
+          def command
+            'dawn -C --json .'
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/dawnscanner/runner.rb]: -C?

File: lib/inquisition/auditors/backend/fasterer/formatter.rb [hunk 10]
```diff
@@ -0,0 +1,43 @@
+module Inquisition
+  module Auditors
+    module Backend
+      module Fasterer
+        class Formatter < Core::BaseFormatter
+          private
+
+          def build_errors
+            @build_errors ||= Hash[*file_with_errors].map(&method(:build_error)).flatten
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/fasterer/formatter.rb:16]: to_h?
try

File: lib/inquisition/auditors/backend/fasterer/formatter.rb [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+                linterable_type: linterable_type_default,
+                message: prepare_message(error_message),
+                type: type_default,
+                line: line
+              }
+            end
+          end
+
+          def prepare_message(message)
+            message.split(' Occurred').first
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/fasterer/formatter.rb]: string elements to constants

File: lib/inquisition/auditors/backend/reek/formatter.rb [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+          private
+
+          def build_errors
+            @build_errors ||= data.map(&method(:build_error)).flatten
+          end
+
+          def build_error(error)
+            error['lines'].map do |line|
+              {
+                linterable_name: error['source'],
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/reek/formatter.rb]: keys to constant

File: lib/inquisition/auditors/backend/rubocop/formatter.rb [hunk 13]
```diff
@@ -24,20 +20,21 @@ def build_errors
...
 
-          def data
-            @data ||= JSON.parse(@unparsed_data)
+          def file_with_errors
+            @file_with_error = data['files'].select { |file| file['offenses'].any? }
           end
 
-          def file_with_errors
-            @file_with_errors ||= data['files'].select { |file| file['offenses'].any? }
+          def data
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/rubocop/formatter.rb]: rename

File: lib/inquisition/config/vendor_autoload.rb [hunk 14]
```diff
@@ -3,3 +3,5 @@
 require 'thor'
 require 'open3'
 require 'benchmark'
+require 'dry/transaction'
```
Comment (dzemlianoi-double) [inline lib/inquisition/config/vendor_autoload.rb]: remove

File: lib/inquisition/core/base_formatter.rb [hunk 15]
```diff
@@ -1,6 +1,32 @@
...
+        },
+        error_count: {
+          default: 0
+        },
+        total_files: {
+          default: 0
+        }
+      }.freeze
+
       def initialize(unparsed_data, spended_time)
```
Comment (dzemlianoi-double) [inline lib/inquisition/core/base_formatter.rb]: keyword args

File: lib/inquisition/auditors/backend/fasterer/formatter.rb (hunk 10 above)
Comment (za4emnik) [reply lib/inquisition/auditors/backend/fasterer/formatter.rb:16]:
```
["node_modules/actiontext/lib/action_text/plain_text_conversion.rb",
 "Using each_with_index is slower than while loop. Occurred at lines: 21.",
 "node_modules/actiontext/lib/action_text/content.rb",
 "Calling blocks with call is slower than yielding. Occurred at lines: 60, 67."]
```

**Comment by za4emnik** [comment]
Doc: 
https://docs.google.com/spreadsheets/d/1RmlEnx6XSCgPeN6YeHuFEGyOumwFQJLAomz8RJNaHhI/edit?usp=sharing

File: lib/inquisition/auditors/backend/bundler_audit/formatter.rb [hunk 16]
```diff
@@ -0,0 +1,44 @@
+module Inquisition
+  module Auditors
+    module Backend
+      module BundlerAudit
+        class Formatter < Core::BaseFormatter
+          LINTERABLE_NAME_KEY = 'Name'.freeze
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/formatter.rb]: unify constants

File: lib/inquisition/auditors/backend/bundler_audit/formatter.rb [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+                linterable_name: vulnerability[LINTERABLE_NAME_KEY],
+                linterable_type: linterable_type_default,
+                message: vulnerability[MESSAGE_KEY],
+                type: vulnerability[TYPE_KEY],
+                line: line_default
+              }
+            end
+          end
+
+          def hash_data
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/formatter.rb]: raw_errors

File: lib/inquisition/auditors/backend/bundler_audit/formatter.rb [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+            @hash_data ||= split_raw_data
+          end
+
+          def split_raw_data
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/bundler_audit/formatter.rb]: raw_errors

File: lib/inquisition/auditors/backend/dawnscanner/formatter.rb [hunk 19]
```diff
@@ -0,0 +1,39 @@
+module Inquisition
+  module Auditors
+    module Backend
+      module Dawnscanner
+        class Formatter < Core::BaseFormatter
+          MESSAGE_KEY = 'message'.freeze
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/dawnscanner/formatter.rb]: fix

File: lib/inquisition/auditors/backend/dawnscanner/formatter.rb [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+          TYPE_KEY = 'severity'.freeze
+          ERROR_COUNT_KEY = 'vulnerabilities_count'.freeze
+
+          private
+
+          def build_errors
+            hash_data['vulnerabilities'].map do |vulnerability|
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/dawnscanner/formatter.rb]: separate constant or key of some another part(Hash) constant

File: lib/inquisition/auditors/backend/reek/formatter.rb [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+          TYPE_KEY = 'smell_type'.freeze
+
+          private
+
+          def build_errors
+            @build_errors ||= hash_data.map(&method(:build_error)).flatten
+          end
+
+          def build_error(error)
+            error['lines'].map do |line|
```
Comment (dzemlianoi-double) [inline lib/inquisition/auditors/backend/reek/formatter.rb]: separate key

**Review by dzemlianoi-double — APPROVED** [review]

**Pull request merged**